lib/stripe.js

```diff
@@ -227,7 +231,11 @@ Stripe.prototype = {
   },
 
   setHttpAgent: function(agent) {
-    this._setApiField('agent', agent);
+    if (agent instanceof https.Agent) {
+      this._setApiField('https_agent', agent);
+    } else {
+      this._setApiField('http_agent', agent);
```

Comment:
I like this approach. Glad to see the docs already tell you to pass an https (not http) agent to this.